docs: add metadata to all remaining pages (#1246)
canonical/rockcraft: docs/reference/rockcraft-yaml.rst

.. meta::

>     :description: Reference documentation for all keys in the rockcraft.yaml project file. Covers name, title, summary, base, build-base, parts, services, and all options.

Reviewer: ```suggestion
    :description: Configuration reference for all keys in rockcraft.yaml.
```
